data/docs/messaging-queues/kafka.mdx

```diff
@@ -415,9 +415,417 @@ Run the collector with the above configuration :
+                regex: Kafka
+                action: keep
+```
+
+Deploy the collector:
+```bash
+helm install my-release signoz/k8s-infra -f ./signoz-k8.yml
+```
+
+### Step 5: Deploy Producer and Consumer Applications
+
+Deploy the Kafka producer and consumer applications with OpenTelemetry instrumentation:
+
+**Compile your Java producer and consumer applications:** Ensure your producer and consumer apps are compiled and ready to run.
+
+**Run Producer App with Java Agent:**
+
+```bash
+java -javaagent:/path/to/opentelemetry-javaagent.jar \
+     -Dotel.service.name=producer-svc \
+     -Dotel.traces.exporter=otlp \
+     -Dotel.metrics.exporter=otlp \
+     -Dotel.logs.exporter=otlp \
+     -jar /path/to/your/producer.jar
+```
+
+**Run Consumer App with Java Agent:**
+
+```bash
+java -javaagent:/path/to/opentelemetry-javaagent.jar \
+     -Dotel.service.name=consumer-svc \
+     -Dotel.traces.exporter=otlp \
+     -Dotel.metrics.exporter=otlp \
+     -Dotel.logs.exporter=otlp \
+     -Dotel.instrumentation.kafka.producer-propagation.enabled=true \
+     -Dotel.instrumentation.kafka.experimental-span-attributes=true \
+     -Dotel.instrumentation.kafka.metric-reporter.enabled=true \
+     -jar /path/to/your/consumer.jar
+```
+
+### Example: Deploying Sample Kafka Producer and Consumer Applications with OpenTelemetry Java Agent
+
+```yaml
+# Producer Deployment
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kafka-producer1
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kafka-producer1
+  template:
+    metadata:
+      labels:
+        app: kafka-producer1
+    spec:
+      containers:
+      - name: producer
+        image: jaikanthjay46/kafka-consumer-test:latest
+        env:
+        - name: BOOTSTRAP_SERVERS
+          value: "signoz-demo-cluster1-kafka-bootstrap:9092"
+        - name: TOPIC
+          value: "topic1"
+        - name: PARTITION_KEY
+          value: "key1"
+        - name: DELAY
+          value: "10"
+        - name: OTEL_SERVICE_NAME
+          value: "producer-svc1"
+        - name: OTEL_TRACES_EXPORTER
+          value: "otlp"
+        - name: OTEL_METRICS_EXPORTER
+          value: "otlp"
+        - name: OTEL_LOGS_EXPORTER
+          value: "otlp"
+        - name: OTEL_EXPORTER_OTLP_LOGS_ENDPOINT
+          value: "http://my-release-k8s-infra-otel-agent:4318/v1/logs"
+        - name: OTEL_EXPORTER_OTLP_METRICS_ENDPOINT
+          value: "http://my-release-k8s-infra-otel-agent:4318/v1/metrics"
+        - name: OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
+          value: "http://my-release-k8s-infra-otel-agent:4318/v1/traces"
+---
+# Consumer Deployment
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kafka-consumer1
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kafka-consumer1
+  template:
+    metadata:
+      labels:
+        app: kafka-consumer1
+    spec:
+      containers:
+      - name: consumer
+        image: jaikanthjay46/kafka-consumer-test:latest
```

Comment:
Same as above for producer. Please add a note mentioning explicitly that the user has to change this with their own docker image.